flutter/lib/src/feedback/sentry_feedback_widget.dart

```diff
@@ -0,0 +1,266 @@
+    this.isNameRequired = false,
+    this.isEmailRequired = false,
+  }) : assert(associatedEventId != const SentryId.empty());
+
+  final SentryId? associatedEventId;
+  final Hub? hub;
+
+  final String title;
+
+  final String nameLabel;
+  final String namePlaceholder;
+  final String emailLabel;
+  final String emailPlaceholder;
+  final String messageLabel;
+  final String messagePlaceholder;
+
+  final String submitButtonLabel;
+  final String cancelButtonLabel;
+
+  final String isRequiredLabel;
+
+  final bool isNameRequired;
+  final bool isEmailRequired;
+
+  @override
+  _SentryFeedbackWidgetState createState() => _SentryFeedbackWidgetState();
+}
+
+class _SentryFeedbackWidgetState extends State<SentryFeedbackWidget> {
+  final TextEditingController _nameController = TextEditingController();
+  final TextEditingController _emailController = TextEditingController();
+  final TextEditingController _messageController = TextEditingController();
+
+  String? _name;
+  String? _email;
+  String? _message;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text(widget.title),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          children: [
+            Expanded(
+              child: SingleChildScrollView(
+                child: Column(
+                  crossAxisAlignment: CrossAxisAlignment.start,
+                  mainAxisSize: MainAxisSize.min,
+                  children: [
+                    Row(
+                      children: [
+                        Text(
+                          key: const ValueKey('sentry_feedback_name_label'),
+                          widget.nameLabel,
+                          style: Theme.of(context).textTheme.labelMedium,
+                        ),
+                        const SizedBox(width: 4),
+                        if (widget.isNameRequired)
+                          Text(
+                            key: const ValueKey(
+                                'sentry_feedback_name_required_label'),
+                            widget.isRequiredLabel,
+                            style: Theme.of(context).textTheme.labelMedium,
+                          ),
+                      ],
+                    ),
+                    const SizedBox(height: 4),
+                    TextField(
+                      key: const ValueKey('sentry_feedback_name_textfield'),
+                      style: Theme.of(context).textTheme.bodyLarge,
+                      decoration: InputDecoration(
+                        border: const OutlineInputBorder(),
+                        hintText: widget.namePlaceholder,
+                        errorText: _errorText(_name),
+                      ),
+                      controller: _nameController,
+                      keyboardType: TextInputType.text,
+                      onChanged: (text) => setState(() => _name = text),
```

Comment:
The `setState` calls are not necessary, since `_name` and friends aren't displayed to the user at all.

Reply:
This is needed for validation. At first, we don't show validation errors, as everything is empty. After tapping submit once, validation issues are shown inline and updated as the user types.

Reply:
It's not needed with https://github.com/getsentry/sentry-dart/pull/2240#discussion_r1723252999, but yeah, the current approach needs it.